### PR TITLE
formula: add std_go_args

### PR DESCRIPTION
**Pull request by dawidd6**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Discussion [here](https://github.com/Homebrew/brew/pull/6923) and [here](https://github.com/Homebrew/homebrew-core/issues/47627).

### CODE REVIEW

**Comment by alebcay** [comment]
I merged a good number of the formulae in https://github.com/Homebrew/homebrew-core/issues/47627, so just wanted to offer some feedback/input.
- Although I initially endorsed `-ldflags "-s -w"`, I'm now less confident about it. While it does shrink binary size considerably, it breaks a number of formulae. With that being said, the breakages are fairly obvious (blatant test fail or the resulting binary is not a valid executable). So I would propose that `-ldflags "-s -w"` be in `std_go_build_args` instead if we want to include it, and just omit the inclusion of it in any formulae for which the `ldflags` break things. Another option of course is simply to omit these flags altogether and just live with bigger binaries and possibly less headache.
- `-trimpath` seems to be a pretty reliable pick so far. Others mentioned on the issue in core that Debian uses it, and it hasn't broken anything so far.

**Comment by MikeMcQuaid** [comment]
> * Another option of course is simply to omit these flags altogether and just live with bigger binaries and possibly less headache.

I'm 👍 on this.

> * `-trimpath` seems to be a pretty reliable pick so far. Others mentioned on the issue in core that Debian uses it, and it hasn't broken anything so far.

This also sounds good to me. If I've understood correctly it should also result in more `cellar :any` bottles.

**Comment by dawidd6** [comment]
So, only `-trimpath` and `-o` options then.

File: Library/Homebrew/formula.rb [hunk 1]
```diff
@@ -1338,6 +1338,11 @@ def std_cmake_args
     args
   end
 
+  # Standard parameters for Go builds.
+  def std_go_args
+    ["-trimpath", "-o", bin/name]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula.rb]: Presumably if you pass another `-o` here it'll be used instead?

Comment (dawidd6) [reply Library/Homebrew/formula.rb]: Answered here: https://github.com/Homebrew/brew/pull/6923#discussion_r367653075

**Comment by MikeMcQuaid** [comment]
Thanks again @dawidd6!

**Pull request merged**